src/main/java/co/rsk/federate/btcreleaseclient/cache/PegoutSignedCacheImpl.java

```diff
@@ -41,14 +51,19 @@ public void putIfAbsent(Keccak256 pegoutCreationRskTxHash) {
         .ifPresent(rskTxHash -> cache.putIfAbsent(rskTxHash, Instant.now()));
   }
 
+  void performCleanup() {
+    cache.entrySet().removeIf(
+        entry -> !hasTimestampNotExpired(entry.getValue()));
+  }
+
   private boolean hasTimestampNotExpired(Instant timestampInCache) {
     return Optional.ofNullable(timestampInCache)
         .map(timestamp -> Instant.now().toEpochMilli() - timestamp.toEpochMilli())
         .map(timeCachedInMillis -> timeCachedInMillis <= ttl.toMillis())
         .orElse(false);
   }
 
-  private static void assertValidTtl(Duration ttl) {
+  private Duration validateTtl(Duration ttl) {
```

Comment:
Don't return the same `ttl` value. This method can be `void`. It can also be `static`, since looks more like an utility method and doesn't access any instance fields.

Reply:
Sounds good, I have updated to what I had before.